### ISSUE TITLE
First Assigment

### ISSUE THREAD

**Issue by snowfrogdev** [Neoflash1979/learn-gamedev#2]
We are looking for some ideas for the very first assignment. Looking forward to your suggestions.

**Comment by GamerTurtle** [comment]
Could I suggest something about starting small? 
I imagine a short document about what the requirements of different games would be, how many assets, what technical components and then a quiz along the lines of:
Q: What is required for a platformer
A: Collision checks, Save games, Level selection, Pickup objects
Q: What is required for an open world RPG
A: Character skills, item management, fast travel, NPC AI, quests, efficient terrain and buildings, combat system

**Comment by snowfrogdev** [comment]
@GamerTurtle Thanks for your input. Right now, the idea is that each assignment is a complete game development assignment. It will take the form of a simple set of requirements that the game must meet. So, in this issue, we are meant to discuss what kind of game we want the learning game dev to make. We need to keep the scope small, like you said.

**Comment by adam-street** [comment]
I think making it in the form of a game jam would be a good idea.
Limit the time.
Give them a keyword to build there ideas from.
Add a small set of loose requirement the game must meet.

You could then provide tutorials on the requirements allowing the devs to feel free to be creative without feeling lost.

**Comment by marnel-estrada** [comment]
I think the first assignment should be a text based game. A simple RPG battle with decisions like Attack, Defend, Use Item could go a long way to understanding programming.

**Comment by snowfrogdev** [comment]
@marnel-estrada I tend to agree. I also think that the first game should be pretty much entirely text based. I think the focus of the first assignment should be on learning how to use whatever engine/ programming language the learner chooses to use.

@adam-street I also agree with time boxing the assignments. I think this point is crucial because one of the hardest things to do for beginning devs is to keep the scope small. Time boxing the assignments will prevent feature creep and keep the learner focused and progressing thru the assignments.
